### DISCUSSION TITLE
How to include java deps when compiling idris files

### DISCUSSION THREAD

**Discussion by jneira** [mmhelloworld/idris-jvm#181] [Q&A] [answered]
Hi, i am experimenting with idris-jvm and my next step is make ffi bindings for java libraries.
I've been investigating https://github.com/mmhelloworld/idris-spring-boot-example and i guess includind deps was done here: 
https://github.com/mmhelloworld/idris-spring-boot-example/blob/main/build.gradle#L45C1-L48

But that example uses idris 1 and there is no type providers for idris2 afaik.
What would be the new way to include them?

Thanks!

**Comment by mmhelloworld** [comment, marked as answer]
Hi,

To provide FFI bindings, Java libraries need not be excluded for compilation. Idris JVM compiler doesn't need any class files or Java classpath to be setup to generate bytecode. It is only at the runtime that the FFI Java dependencies need to be provided. Type Providers in Idris 1 were mainly used to auto generate FFI bindings which required the classes to be in classpath as the types were generated using reflection. In Idris 2, FFI bindings need to be provided explicitly for now so the compiler doesn't need the actual classpath.

I just pushed new changes for [idris-spring-boot-example](https://github.com/mmhelloworld/idris-spring-boot-example) to run on Idris 2. If you see [here in Maven build file](https://github.com/mmhelloworld/idris-spring-boot-example/blob/main/pom.xml#L148), compiler is invoked only passing `ipkg` file. Then Maven dependency plugin [copies](https://github.com/mmhelloworld/idris-spring-boot-example/blob/main/pom.xml#L102) all the runtime dependencies into the output folder.

**Reply by jneira** [reply]
makes sense, thanks for the info
